tsfresh/utilities/dataframe_functions.py

```diff
@@ -440,57 +528,76 @@ def roll_time_series(df_or_dict, column_id, column_sort, column_kind, rolling_di
     else:
         grouper = [column_id, ]
 
-    if column_sort is not None and df[column_sort].dtype != np.object:
-
+    if column_sort is not None:
         # Require no Nans in column
         if df[column_sort].isnull().any():
             raise ValueError("You have NaN values in your sort column.")
 
         df = df.sort_values(column_sort)
 
-        # if rolling is enabled, the data should be uniformly sampled in this column
-        # Build the differences between consecutive time sort values
+        if df[column_sort].dtype != np.object:
+            # if rolling is enabled, the data should be uniformly sampled in this column
+            # Build the differences between consecutive time sort values
 
-        differences = df.groupby(grouper)[column_sort].apply(
-            lambda x: x.values[:-1] - x.values[1:])
-        # Write all of them into one big list
-        differences = sum(map(list, differences), [])
-        # Test if all differences are the same
-        if differences and min(differences) != max(differences):
-            warnings.warn("Your time stamps are not uniformly sampled, which makes rolling "
-                          "nonsensical in some domains.")
+            differences = df.groupby(grouper)[column_sort].apply(
+                lambda x: x.values[:-1] - x.values[1:])
```

Comment:
I could (actually, I copied this from the code before - but still a valid comment) - but this would change the types and would therefore a bit more `Series` to `list` conversion stuff. I would prefer to keep it for now.